src/wtforms/fields/core.py

```diff
@@ -1104,6 +1108,8 @@ def __init__(
         self.max_entries = max_entries
         self.last_index = -1
         self._prefix = kwargs.get("_prefix", "")
+        self._separator = separator
+        self._field_separator = unbound_field.kwargs.get("separator", "-")
```

Comment:
Is there a time where _separator and _field_separator would diverge in value?

Reply:
Yes, see `test_enclosed_subform_mixed_separators` for example. When assigning a separator to the contained form field but not to the containing list these fields diverge.